### PR TITLE
Update load_bsi.py

### PR DESCRIPTION
**Pull request by EmmaErwin93**
Updated for new variable names acquired with BSI update (Applied for UMAINE radar in spring 2023)

### CODE REVIEW

**Comment by dlilien** [comment]
@EmmaErwin93 I modified this so that it hopfully works simultaneously with old and new BSI data. I have confirmed that it works with old data again. Can you check that this version works with your new BSI data? I think you just need to `git pull` your fork to get my changes.

**Comment by EmmaErwin93** [comment]
@dlilien Just tried, but get an error:

Loading LabVIEW Boolean from raw/Eclipse-P2/eclipse-phase2.h5
Loading line_0 from raw/Eclipse-P2/eclipse-phase2.h5
Interpolating traces with bad GPS in line_0
Traceback (most recent call last):
  File "/Users/emmaerwin/opt/miniconda3/bin/impdar", line 33, in <module>
    sys.exit(load_entry_point('impdar', 'console_scripts', 'impdar')())
  File "/Users/emmaerwin/software/ImpDAR/impdar/bin/impdarexec.py", line 182, in main
    return args.func(**vars(args))
  File "/Users/emmaerwin/software/ImpDAR/impdar/lib/load/__init__.py", line 196, in load_and_exit
    rd_list = load(filetype, fn_i, channel=channel, t_srs=t_srs, s_srs=s_srs, *args, **kwargs)
  File "/Users/emmaerwin/software/ImpDAR/impdar/lib/load/__init__.py", line 92, in load
    data_nestedlist = [load_bsi.load_bsi(fn, nans=nans) for fn in fns_in]
  File "/Users/emmaerwin/software/ImpDAR/impdar/lib/load/__init__.py", line 92, in <listcomp>
    data_nestedlist = [load_bsi.load_bsi(fn, nans=nans) for fn in fns_in]
  File "/Users/emmaerwin/software/ImpDAR/impdar/lib/load/load_bsi.py", line 180, in load_bsi
    time = interp1d(h5_data.trace_num[mask],
  File "/Users/emmaerwin/opt/miniconda3/lib/python3.9/site-packages/scipy-1.9.3-py3.9-macosx-10.9-x86_64.egg/scipy/interpolate/_interpolate.py", line 484, in __init__
    self._y = self._reshape_yi(self.y)
  File "/Users/emmaerwin/opt/miniconda3/lib/python3.9/site-packages/scipy-1.9.3-py3.9-macosx-10.9-x86_64.egg/scipy/interpolate/_polyint.py", line 110, in _reshape_yi
    return yi.reshape((yi.shape[0], -1))
ValueError: cannot reshape array of size 0 into shape (0,newaxis)

File: impdar/lib/load/load_bsi.py [hunk 1]
```diff
@@ -159,9 +160,9 @@ def load_bsi(fn_h5, XIPR=True, channel=0., line=None, nans=None, *args, **kwargs
                     h5_data.elev[location_num] = np.nan
 
             h5_data.dt = 1.0 / float(
-                _xmlGetVal(digitizer_data, sample_rate_str))
```
Comment (EmmaErwin93) [inline impdar/lib/load/load_bsi.py:162]: I think I forgot to adjust this:
 _xmlGetVal(digitizer_data, 'Sample Rate'))

**Comment by dlilien** [comment]
Ok, @EmmaErwin93 can you send me a file that is failing to load (or if you figured it out then just push the fix)? Sorry that this is complicated, but I know people are using ImpDAR with both old and new versions of BSI systems, so we need to provide some kind of solution that works for both.

**Comment by dlilien** [comment]
This should be compatible with both versions now. @EmmaErwin93 let me know if there are any lingering problems.

**Comment by EmmaErwin93** [comment]
@dlilien  I think there is an error in line 163 of the load_bsi.py file:         _xmlGetVal(digitizer_data, sample_rate_str) or 1.0)

should not have the "or 1.0" at the end -- I think i accidentally hard coded this for a test file I was doing, but it affects the time vector for any file that the sample_rate_str is not read in. 

I think that sample_rate_str in line 103 needs to have a space in front: ' Sample Rate'

**Pull request merged**